### PR TITLE
#9432 [DFL] Aside block styling fixes 

### PR DESCRIPTION
**Pull request by stevedya**
# Description

This PR addresses button style issues for #9432 as well as adds a new streamfield for allowing blocks with asides

**Modifications made:**

- Left align aside button
- Add tailwindcss/container-queries plugin (this is used to determine when the container has gotten smaller by the CTA sidebar or signup form and renders the aside in a single column above the left content side). This is needed because when the CTA or signup form is on desktop we can tell how wide the left column is and be responsive with the content to avoid the aside rendering on the same row. 

### Example when paragraph is used  with no aside and listing block is used with an aside
Screen Shot 2022-10-28 at 12 44 43 PM

### How this content looks in wagtail
Screen Shot 2022-10-28 at 12 42 27 PM

### Before entering in content in wagtail
Screen Shot 2022-10-28 at 12 41 25 PM

### Example when cta or signup form is on the page
Screen Shot 2022-10-27 at 10 50 33 PM

Related PRs/issues:  #9432, #9463, #9393

# Checklist

**Tests**
- [ ] ~~Is the code I'm adding covered by tests?~~

**Changes in Models:**
- [ ] ~~Did I update or add new fake data?~~
- [ ] ~~Did I squash my migration?~~
- [x] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [ ] ~~Is my code documented?~~
- [ ] ~~Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

**Review by tbrlpld — CHANGES_REQUESTED** [review]
Thanks for giving this a shot @stevedya. While it does work in Chrome and Safari, the container queries are not yet supported in Firefox, which I think we need before using something on this site.

https://caniuse.com/css-container-queries

**Comment by stevedya** [comment]
@tbrlpld Thats alright, the container queries aren't a make or break for this to work properly. This layout will still work with regular media queries as well. 

**Note:** The only issue that the container queries were solving was for the instances when you have a CTA or Signup block active on the site (causes the content to be broken into its narrow width layout and this block on desktop would still be in its 3 column layout because its relying on the screen size). 

So another option i forgot to consider, which is arguably a cleaner approach is to add a parent class when the CTA or Signup is active, that we can use to set this block to a single column. 

I'll quickly modify it to work this way

**Comment by stevedya** [comment]
@tbrlpld This one is ready for you to look at again 😄

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class BlockWithAside(blocks.StructBlock):
+    content = blocks.StreamBlock([
+        ("listing", ListingBlock()),
+        ('paragraph', blocks.RichTextBlock(
+            features=full_content_rich_text_options
+        ))
+    ], max_num=1, help_text='The wider block that appears on the left on desktop', icon='doc-full')
+
+    aside = blocks.StreamBlock(aside_fields, required=False, icon='doc-full', help_text='Elements here will appear in the column on the right side of the page on desktop. This can be left blank if you would just like a 2/3 column on the left')
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py:15]: What do you think about adding some line breaks here?

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+from .full_content_rich_text_options import full_content_rich_text_options
+
+
+class BlockWithAside(blocks.StructBlock):
+    content = blocks.StreamBlock([
+        ("listing", ListingBlock()),
+        ('paragraph', blocks.RichTextBlock(
+            features=full_content_rich_text_options
+        ))
+    ], max_num=1, help_text='The wider block that appears on the left on desktop', icon='doc-full')
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py:13]: What do you thing about using the same indentation for all the attributes? I think that is a bit more common. 


I know we need to set up a formatter for this repo ... 😓

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/block_with_aside.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  [&.block-with-aside_+_.block-with-aside] - Will add a margin to space out these specific blocks from each other
+  The blocks inside content will determine their own margin bottom for vertical spacing
+
+  group-[.has-sidebar] - If the parent has a sidebar like CTA or Signup Block then apply styles to convert this into a single column
+{% endcomment %}
+
+<section class="block-with-aside [&.block-with-aside_+_.block-with-aside]:tw-mt-7">
+  <div class="tw-grid tw-grid-cols-1 large:tw-grid-cols-3 tw-gap-4 large:tw-gap-7 group-[.has-sidebar]:tw-grid-cols-1 group-[.has-sidebar]:tw-gap-4 ">
+    <div class="tw-col-span-1 large:tw-col-span-2 group-[.has-sidebar]:tw-col-span-1">
+      {{ value.content }}
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/block_with_aside.html:11]: This works 😲 I would have suspected we need `{% include_block %}` or something

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/listing_block.html [hunk 4]
```diff
@@ -2,26 +2,29 @@
...
-  <div class="tw-row tw-py-4">
+  <ul class="tw-mb-7 tw-list-none tw-pl-0 tw-grid tw-gap-4 medium:tw-grid-cols-2 medium:tw-gap-7">
     {% for block in self.cards %}
       {% with card=block %}
         {% image card.image fill-350x197 as img %}
-        <div class="tw-px-4 medium:tw-w-1/2 tw-mb-5 group">
+        <li>
+          {% if card.link %}
+            <a href="{% pageurl card.link %}"
+               class="tw-group/listing hover:tw-no-underline focus:tw-no-underline hover:tw-text-black focus:tw-text-black"
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/listing_block.html:12]: Nice. Just learned about named groups yesterday 😅

**Review by tbrlpld — COMMENTED** [review]
I am not 100% sure, but I think the aside should sit outside of the normal width of the document. So basically in the same area where the cta/signups are shown. Like adding an aside should have the same effect as a cta/signup. 🤔 Hm... I am not sure anymore. I think this case has not been designed for. 

The body content should all have the same width, I think. 

Screen Shot 2022-10-28 at 15 47 24

**Comment by cdanfon** [comment]
Let's put this onto staging and ask for Kristina's opinion @tbrlpld

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py (hunk 1 above)
Comment (stevedya) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py]: Yep thats a great idea! I've made this change now

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py (hunk 2 above)
Comment (stevedya) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/block_with_aside.py]: I've pushed the attributes to a new line now so they can be at the same indent, hopefully it's closer to what you were expecting now.

**Comment by tbrlpld** [comment]
@stevedya There also seems to be an unintended change of the narrow body width that is caused by this PR. See the visual regression tests: https://percy.io/Mozilla-Foundation/foundation.mozilla.org/builds/22591214/changed/1259054366?browser=chrome&browser_ids=30%2C31&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

**Comment by stevedya** [comment]
Hey @tbrlpld thanks for noticing those regressions, i've fixed them now so you can check this one out again. 🥲

**Review by tbrlpld — APPROVED** [review]

**Pull request merged**